### ISSUE TITLE
Trouble with reproducing result

### ISSUE THREAD

**Issue by ductuantruong** [wngh1187/ExU-Net#3]
Hi, 

Thank you for making the code & pretrained models public. I am trying to reproduce the results. However, the result I got after running your **pretrained ExU-Net** model on VoxCeleb1-O is different from the paper's result. My reproduce result is `EER_clean: 0.22067868504771956`. To get the result, I just commented the `train` step and load the pretrained model in the `test` step. I am asking whether you have encountered this problem?

Once again, thank you for sharing the code. I am looking forward to hear from you!

**Comment by wngh1187** [comment]
Thank you for bringing this issue to our attention, and we apologize for any inconvenience caused. We have tested the ExU-Net model on VoxCeleb1-O with the method you described, and we were able to reproduce the results as reported in our paper.

For loading the pre-trained model, we added the following code at line 140 in main.py:
"checkpoint = torch.load('./weights/ExU-Net.pt', map_location=args['device'])"
"trainer.model.load_state_dict(checkpoint)"
In addition, we commented out "self.train(epoch)" in 30 line of trainer.py, and used our evaluation method to obtain the following results: EER_clean: 0.02757158006407247, Min_DCF_clean: 0.16362672322375393.

We suggest that you try our modified approach and evaluate the model again. Please let us know if you have any further questions or concerns.

Thank you for your interest in our work, and we look forward to hearing from you soon.

**Comment by ductuantruong** [comment]
Thank you for your fast response. I have tried your code modification, but I still cannot get the similar result, mine is  `EER_clean: 0.20349946977730635`. 

I also have run the training of your `U-Net`. While the loss value does decrease over time, EER value fluctuate around `0.18`. Here is my last 10 lines of the loss log in `loss.txt`:
```
[122412] loss: 1.7106375694274902
[122462] loss: 1.7007594108581543
[122558] loss: 1.6928579807281494
[122608] loss: 1.7107104063034058
[122658] loss: 1.6798439025878906
[122708] loss: 1.7123098373413086
[122804] loss: 1.694210410118103
[122854] loss: 1.687063455581665
[122904] loss: 1.6687785387039185
[122954] loss: 1.7010258436203003
```
Here is my last 10 lines of the EER_clean log in `EER_clean.txt`:
```
EER_clean: 0.18881230116601733
EER_clean: 0.18849416755037113
EER_clean: 0.19241781548255543
EER_clean: 0.19008483563100959
EER_clean: 0.19209968186638388
EER_clean: 0.19151643690365103
EER_clean: 0.19072110286323188
EER_clean: 0.1916755037115547
EER_clean: 0.19141039236478782
EER_clean: 0.1908801696712625
```
I am trying the narrow down where is the problem. I want to ask you whether your loss value in the end of training `U-Net` is similar to mine. If so, the problem may come from calculating EER value. 

Thank you for spending time to support me!

**Comment by ductuantruong** [comment]
Hi, I have been able to achieve your result by using the same python environment as yours. My old env consisting of:
- python: 3.8 
- torch: 1.11.0+cu113
- torchaudio: 0.11.0+cu113
However, I sill don't know why my old env cannot achieve the same result.

**Comment by wngh1187** [comment]
I am pleased to hear that you were able to reproduce my experiment in your environment. 
I have encountered similar issues with experiment results and reproducibility when the environment changes, not only with this model but with others as well. 
This is because internal operations and randomness within Torch and other libraries may differ under different environmental conditions. 
Although not entirely certain, I hope this information serves as a helpful reference for you.

**Comment by ductuantruong** [comment]
Thank you for your help! I appreciate that!

**Comment by chen11111010** [comment]
Excuse me, can I reproduce the results just by changing the environment?

**Comment by ductuantruong** [comment]
Hi chen11111010,

I think you can reproduce the test result with their given pretrained model. However, I am sure you can reproduce the same result if you re-train the model because I am still unable to obtain the same result by re-training
